src/ffi.rs

```diff
@@ -142,22 +150,379 @@ extern "C" {
+                unsafe { Box::from_raw(previous_ptr as *mut _) };
+        }
+        unsafe {
+            GEOSContext_setNoticeMessageHandler_r(self.ptr, page_func, nf_data);
+        }
+    }
+
+    /// Allows to set an error message handler.
+    ///
+    /// Passing [`None`] as parameter will unset this callback.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::GContextHandle;
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    ///
+    /// context_handle.set_error_message_handler(Some(Box::new(|s| println!("new message: {}", s))));
+    /// ```
+    pub fn set_error_message_handler<'a>(&'a self, ef: Option<Box<dyn Fn(&str) + 'a>>) {
+        let ef_data: Box<Option<Box<dyn Fn(&str) + 'a>>> = Box::new(ef);
+
+        unsafe extern "C" fn message_handler_func<'a>(message: *const c_char, data: *mut c_void) {
+            let callback: &Option<Box<dyn Fn(&str) + 'a>> = &*(data as *mut _);
+
+            let bytes = slice::from_raw_parts(message as *const u8, strlen(message));
+            if let Some(ref callback) = *callback {
+                let s = CStr::from_bytes_with_nul_unchecked(bytes);
+                callback(s.to_str().expect("invalid CStr -> &str conversion"))
+            } else {
+                panic!("cannot get closure...")
+            }
+        }
+
+        let page_func = if ef_data.is_some() {
+            Some(message_handler_func as _)
+        } else {
+            None
+        };
+        let ef_data = Box::into_raw(ef_data) as *mut _;
+        let previous_ptr = self.notice_message.replace(ef_data);
+        if !previous_ptr.is_null() {
+            // We free the previous closure.
+            let _callback: Box<Option<Box<dyn Fn(&str) + 'a>>> =
+                unsafe { Box::from_raw(previous_ptr as *mut _) };
+        }
+        unsafe {
+            GEOSContext_setErrorMessageHandler_r(self.ptr, page_func, ef_data);
+        }
+    }
+
+    /// Gets WKB output dimensions.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, Dimensions};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    ///
+    /// context_handle.set_wkb_output_dimensions(Dimensions::TwoD);
+    /// assert!(context_handle.get_wkb_output_dimensions() == Dimensions::TwoD);
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn get_wkb_output_dimensions(&self) -> Dimensions {
+        Dimensions::from(unsafe { GEOS_getWKBOutputDims_r(self.ptr) })
+    }
+
+    /// Sets WKB output dimensions.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, Dimensions};
+    ///
+    /// let context_handle = match GContextHandle::init() {
```

Comment:
by curiosity, why don't you `.expect` here ?

Reply:
More easy to read but that's just my point of view. I can change it if you prefer?

Reply:
nop, I was just wondering, both are fine